File: common/src/main/java/com/ssblur/scriptor/registry/words/DiscountDescriptors.java
package com.ssblur.scriptor.registry.words;

import com.ssblur.scriptor.api.word.Descriptor;
import com.ssblur.scriptor.word.descriptor.discount.*;

@SuppressWarnings("unused")
public class DiscountDescriptors {
  private static final WordRegistry INSTANCE = WordRegistry.INSTANCE;

  public final Descriptor BLOOD_COST = INSTANCE.register("blood_cost", new BloodCostDescriptor());
  public final Descriptor CHEAP = INSTANCE.register("cheap", new CheapDescriptor());
  public final Descriptor HEALTHY = INSTANCE.register("healthy", new HealthyDescriptor());
  public final Descriptor POISONED = INSTANCE.register("poisoned", new PoisonDescriptor());
  public final Descriptor WEAKENED = INSTANCE.register("weakened", new WeakDescriptor());
  public final Descriptor CRITICAL = INSTANCE.register("critical", new CriticalDescriptor());
  public final Descriptor NIGHT = INSTANCE.register("night", new NightDiscountDescriptor());
  public final Descriptor RAIN = INSTANCE.register("rain", new RainDiscountDescriptor());
}

File: common/src/main/kotlin/com/ssblur/scriptor/registry/words/DiscountDescriptors.kt
package com.ssblur.scriptor.registry.words

import com.ssblur.scriptor.registry.words.WordRegistry.register
import com.ssblur.scriptor.word.descriptor.discount.*

@Suppress("unused")
object DiscountDescriptors {
    val BLOOD_COST = register("blood_cost",
        BloodCostDescriptor()
    )
    val CHEAP = register("cheap", CheapDescriptor())
    val HEALTHY = register("healthy",
        HealthyDescriptor()
    )
    val POISONED = register("poisoned",
        PoisonDescriptor()
    )
    val WEAKENED = register("weakened",
        WeakDescriptor()
    )
    val ON_FIRE = register("on_fire",
        OnFireDescriptor()
    )
    val CRITICAL = register("critical",
        CriticalDescriptor()
    )
    val NIGHT = register("night",
        NightDiscountDescriptor()
    )
    val RAIN = register("rain",
        RainDiscountDescriptor()
    )
    val CLEAR_SKIES = register("clear_skies",
        ClearDiscountDescriptor()
    )
}
